### ISSUE TITLE
{strip}..{/strip} strip spaces between words in this case

### ISSUE THREAD

**Issue by yourchoice** [smarty-php/smarty#177]
```
{strip}
    <h1>Text <em>italic</em></h1>
{/strip}
```

This sequence output "Textitalic" without space.

**Comment by yourchoice** [comment]
>>>
```
{strip}
<h1>Text <em>italic</em></h1>
 {/strip} 

```<<<

**Comment by ophian** [comment]
Did you see https://github.com/smarty-php/smarty/commit/a8f8efd4b48b2aa687d4849a1cca7c3ec6d5d922?
Does that fix it or is that the problem, and with which version?

**Comment by yourchoice** [comment]
3.1.30-dev

**Comment by uwetews** [comment]
3.1.30-dev/28 does output it with space as it should at my place

**Comment by yourchoice** [comment]
Hi again,

Yes, seems to be a particular case. I tried to simulate.

This working well (let the space between words):
<h2>Word1 <em>Word2</em></h2>

This not:
<h2>{'Word1'} <em>Word2</em></h2>


For second can by any smarty var.

**Comment by uwetews** [comment]
The fix is now in the master branch and will later be included in 3.1.30

Smarty now does keep the space when the tag does create output.

Just as test case

```php
<h2>{counter} <em>Word2</em></h2>
```
does keep the space
```php
<h2>{counter assign=foo} <em>Word2</em></h2>
```
does not

**Comment by yourchoice** [comment]
Hi, 

Sorry, this issue need reopen.

There exists another case:

...
<i class="fa fa-archive"></i> {#archived_menu#|mb_ucfirst}
...

Same thing, there space is striped.

I think you resolve just for case when a smarty var precedes an html tag
